### DISCUSSION TITLE
Mastodon on three-head failover cluster lose posts

### DISCUSSION THREAD

**Discussion by KdaCMC** [mastodon/mastodon#26495] [Q&A]
Hello,
I run Mastodon on three-head cluster with Pacemaker with two switching 
on reboot nodes. It has Mastodon directory and Postgres DB on DRBD switching to master node. I set up `tootctl feeds build` on Nginx starting alert. This command makes feed better, but I have faced some issues. I subscribed on a guy on another server, and his posts both public and personal to me in "home" appear and disappear according to nodes reboot. In PM I have the chat fully. I have list with only my posts and posts there also depend on current node. Is it possible to make full syncronization between nodes?
Thanks in advance.

**Comment by renchap** [comment]
Mastodon does not have any support for Pacemaker. But as it is using PostgreSQL, it should not loose any data if your setup is done correctly.

Are you sending every query to the current primary server at all time?

**Reply by KdaCMC** [reply]
Pacemaker has support for systemd services and Nginx, so clear there. The PostgreSQL server runs on only one master node with DB on DRBD share, the DRBD directory even is not mounted on the other node. It seems to me the issue is connected with https://docs.joinmastodon.org/admin/migrating/ The  `tootctl feeds build` command is described here near DB migrating, but it does not provide full migrating. I wonder where Mastodon keeps it's cache or something like that.

**Reply by renchap** [reply]
I am not sure I understand why you need to run this command at all.

If you have a working postgres setup, then you do not need to run any command after the primary pg server switched to another.

**Reply by KdaCMC** [reply]
I was thinking the same way, but DB is not enough. From https://docs.joinmastodon.org/admin/migrating/
3. Dump and load the Postgres database using the instructions below.
7. Run `RAILS_ENV=production ./bin/tootctl feeds build` to rebuild the home timelines for each user.

**Reply by renchap** [reply]
You keep linking to the server migration docs, but this is not relevant to what you are trying to achieve. You are not migrating to a new machine, only having the postgresql server migrate to a new server.

You do not need to do any of this, only ensure that your web & sidekiq servers can reach out to the new postgres primary after the failover.

**Reply by KdaCMC** [reply]
In fact it is like migrating to a new machine. Not only PostgreSQL server is switching, the systemd services and Nginx run on different server. The slave node keeps DRBD with DB and Mastodon directory of Nginx synced, but it does not run systemd services or Nginx as well as PostgreSQL.
